src/solvers/dg_tree/dg.jl

```diff
@@ -264,47 +330,66 @@ include("l2projection.jl")
 include("basis_lobatto_legendre.jl")
 
 """
-    DGSEM([RealT=Float64,] polydeg::Integer,
-          surface_flux=flux_central,
-          volume_integral::AbstractVolumeIntegral=VolumeIntegralWeakForm(),
-          mortar=MortarL2(basis))
+    DGSEM(; RealT=Float64, polydeg::Integer,
+            surface_flux=flux_central,
+            surface_integral=SurfaceIntegralWeakForm(surface_flux),
+            volume_integral=VolumeIntegralWeakForm(),
+            mortar=MortarL2(basis))
 
 Create a discontinuous Galerkin spectral element method (DGSEM) using a
 [`LobattoLegendreBasis`](@ref) with polynomials of degree `polydeg`.
 """
-const DGSEM = DG{Basis, Mortar, SurfaceFlux, VolumeIntegral} where {Basis<:LobattoLegendreBasis, Mortar, SurfaceFlux, VolumeIntegral}
+const DGSEM = DG{Basis} where {Basis<:LobattoLegendreBasis}
```

Comment:
Isn't this slightly too restrictive now? What if we want to do work on Legendre Gauss nodes again? We'd lose SBP, but it might be interesting for other reasons. Similar arguments can be made for deprecating the DGSEM ctors that allow to specify the basis.

Reply:
Well, that's what we already had before. We also don't have Gauß-Legendre bases in Trixi right now. In addition, these would require major changes to a lot of methods (they are still SBP, just more complicated).
Thus, we would need to have another name for DGSEM with Legendre-Gauß-Lobatto basis. `DGSEMLGL` doesn't look nice, I think...
What about postponing this until the need to change anything arises? We can always throw this stuff away in Trixi v0.4.

Reply:
No, I absolutely agree. I just interpreted some of the comments you made that you are deprecating something that might be hard to reintroduce. But I hadn't realized that yes, DGSEM was already dependent on a LobattoLegendreBasis, so nothing really changes :+1:

Reply:
Yeah, this line just simplifies what we had before